### ISSUE TITLE
no-ref-object-reactivity-loss: false negative when the accessing is wrapped in IIFE with the same scope

### ISSUE THREAD

**Issue by n0099** [vuejs/eslint-plugin-vue#3056]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 10.0.2
- **eslint-plugin-vue version:** 10.8.0
- **Vue version:** 3.5.29
- **Node version:** 24.12.0
- **Operating System:** NixOS

**Please show your full configuration:**

https://github.com/n0099/open-tbm/blob/e48097c0ac96da264942f52d74bcd294855df35a/fe/eslint.config.ts

**What did you do?**	

```ts
const firstThread = computed(() => firstPostPage.value?.threads[0]); // https://github.com/n0099/open-tbm/blob/92fd7111a62c16f0f7ec29da5fae04a34532f1e2/fe/src/utils/post/seo/index.ts#L18
console.log(firstThread.value?.title);
//          ^ reporting violation of eslint rule `vue/no-ref-object-reactivity-loss`
(() => {
    const firstThreadTitle = computed(() => firstThread.value?.title); // https://github.com/n0099/open-tbm/blob/92fd7111a62c16f0f7ec29da5fae04a34532f1e2/fe/src/utils/post/seo/index.ts#L56
    console.log(firstThread.value?.title);
    //          ^ not reporting violation of eslint rule `vue/no-ref-object-reactivity-loss`
    console.log(firstThreadTitle.value);
    //          ^ reporting violation of eslint rule `vue/no-ref-object-reactivity-loss`
})();
```

**What did you expect to happen?**
`firstThread.value?.title` inside this IIFE should also report.

**What actually happened?**

Not reporting.

**Repository to reproduce this issue**

https://github.com/n0099/open-tbm/blob/e48097c0ac96da264942f52d74bcd294855df35a/fe

**Comment by n0099** [comment]
This also happens when the IIFE is a plain function outside the scope of `firstThread` being defined like:
```ts
const useOgImagePost = (firstThread: Ref<ResponsePage['threads'][number] | undefined>) => {
    console.log(firstThread.value?.title);
    //          ^ not reporting violation of eslint rule `vue/no-ref-object-reactivity-loss`
    console.log(firstThreadTitle.value);
    //          ^ reporting violation of eslint rule `vue/no-ref-object-reactivity-loss`
};
(() => {
    const firstThread = computed(() => firstPostPage.value?.threads[0]);
    useOgImagePost(firstThread);
})();
```

**Comment by ValentinYoushkevich** [comment]
Root cause: the rule pushes a new scope entry on entering **every** function and then compares the scope of the `ref()`/`computed()` call with the current one by identity. The callee of an IIFE gets its own entry, so the same-scope check fails and the report is silenced — even though the body runs synchronously in the enclosing scope and loses reactivity exactly like a direct access.

Traced on the rule itself:

```
node=count defineScope=Program#depth0 currentScope=Program#depth0                same=true   -> reported
node=count defineScope=Program#depth0 currentScope=ArrowFunctionExpression#depth1 same=false  -> silent
```

Opened #3115: the enclosing scope is kept when the entered function is the callee of a call expression, so both forms are now reported:

```js
import { ref } from 'vue'
const count = ref(0)
;(() => {
  const v1 = count.value // reported
})()
;(function () {
  const v2 = count.value // reported
})()
```

`async` functions and generators are deliberately left out — their bodies are not guaranteed to run synchronously to the end, so reporting there would risk false positives.

The second case from your follow-up comment (a ref passed as an argument to a plain function that is called from an IIFE) is not covered by that PR: it needs inter-procedural analysis, which is a much bigger change than this fix.
